thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/onboard/tasks/FunctionCreationOnboardingTask.java

```diff
@@ -105,40 +117,66 @@ public void run() {
     Preconditions.checkNotNull(configuration.getString(FUNCTION_NAME));
     Preconditions.checkNotNull(configuration.getString(COLLECTION_NAME));
     Preconditions.checkNotNull(configuration.getString(METRIC_NAME));
-    Preconditions.checkNotNull(configuration.getString(CRON_EXPRESSION));
-    Preconditions.checkNotNull(configuration.getString(WINDOW_SIZE));
-    Preconditions.checkNotNull(configuration.getString(WINDOW_UNIT));
-    Preconditions.checkNotNull(configuration.getString(PROPERTIES));
     Preconditions.checkArgument(configuration.containsKey(ALERT_ID) || configuration.containsKey(ALERT_NAME));
     if (!configuration.containsKey(ALERT_ID)) {
       Preconditions.checkNotNull(configuration.getString(ALERT_TO));
     }
 
+    // check if duplicate name exists
+    List<AnomalyFunctionDTO> duplicateFunctions = anoomalyFunctionDAO.findWhereNameLike(configuration.getString(FUNCTION_NAME));
+    for (AnomalyFunctionDTO duplicateFunction : duplicateFunctions) {
+      if (duplicateFunction.getFunctionName().equals(configuration.getString(FUNCTION_NAME))) {
+        throw new IllegalArgumentException("Duplicate function name " + configuration.getString(FUNCTION_NAME)
+            + " is found");
+      }
+    }
+
+    if (configuration.getString(ALERT_NAME) != null) {
+      List<AlertConfigDTO> duplicateAlerts = alertConfigDAO.findWhereNameLike(configuration.getString(ALERT_NAME));
+      for (AlertConfigDTO duplicateAlert : duplicateAlerts) {
+        if (duplicateAlert.getName().equals(configuration.getString(ALERT_NAME))) {
+          throw new IllegalArgumentException("Duplicate alert name " + configuration.getString(ALERT_NAME)
+              + " is found");
+        }
+      }
+    }
+
     // update datasetConfig
     DatasetConfigDTO datasetConfig = datasetConfigDAO.findByDataset(configuration.getString(COLLECTION_NAME));
     if (datasetConfig == null) {
       throw new NoSuchElementException("Cannot find collection: " + configuration.getString(COLLECTION_NAME));
     }
     TimeSpec timeSpec = ThirdEyeUtils.getTimeSpecFromDatasetConfig(datasetConfig);
     TimeGranularity dataGranularity = timeSpec.getDataGranularity();
+    TimeGranularity userAssignedDataGranularity = null;
```

Comment:
Should we throw an exception and stops the job if the user gives a bad data granularity?

We could swallow the error and continue, but the user would never know the custom granularity is not used.

Reply:
That is a good thought. We should throw out the exception. I will change that.